### ISSUE TITLE
tsserver: project references are ignored

### ISSUE THREAD

**Issue by ukch** [dense-analysis/ale#3752]
## Information

**VIM version**

NVIM v0.4.4
Build type: Release

Operating System: Debian unstable

## What went wrong

TypeScript since version 3 has supported '[Project References](https://www.typescriptlang.org/docs/handbook/project-references.html)', where you can split a project up into multiple sub-projects, by configuring your `tsconfig.json` a little like this:

```json
{
  ...
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./some.child.tsconfig.json"
    },
    {
      "path": "./some.otherchild.tsconfig.json"
    }
  ]
}
```

In this setup, running `tsc` with no arguments will (apparently for compatibility reasons) ignore the sub-projects. Instead you must run `tsc --build` (or `tsc -b`).

I have noticed in such a project, ALE's tsserver will behave like the `tsc` with no arguments (i.e. will ignore sub-projects and not provide output on any of my files). Can it be configured to somehow use the `--build` option?

## Reproducing the bug

* Have a file with a linting error.
* `tsconfig.json` as above
* `some.child.tsconfig.json` configred to catch the linting error
* Open the problem file in vim
Expected: the error should be caught
Actual: the error is ignored

### :ALEInfo

```

 Current Filetype: typescript
Available Linters: ['eslint', 'standard', 'tslint', 'tsserver', 'typecheck', 'xo']
  Enabled Linters: ['eslint', 'tslint']
 Suggested Fixers: 
  'eslint' - Apply eslint --fix to a file.
  'prettier' - Apply prettier to a file.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'tslint' - Fix typescript files with tslint --fix.
  'xo' - Fix JavaScript/TypeScript files using xo --fix.
 Linter Variables:

let g:ale_typescript_tslint_config_path = ''
let g:ale_typescript_tslint_executable = 'tslint'
let g:ale_typescript_tslint_ignore_empty_files = 0
let g:ale_typescript_tslint_rules_dir = ''
let g:ale_typescript_tslint_use_global = 0
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = v:null
let g:ale_command_wrapper = v:null
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '[%linter%] %s [%severity%]'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = v:null
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {'jsx': 'css'}
let g:ale_linters = {'javascript.jsx': ['stylelint, eslint'], 'graphql': ['gqlint'], 'typescript': ['eslint', 'tslint'], 'python': ['flake8', 'black', 'mypy']}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = v:null
let g:ale_list_window_size = v:null
let g:ale_loclist_msg_format = v:null
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = v:null
let g:ale_max_signs = v:null
let g:ale_maximum_file_size = v:null
let g:ale_open_list = v:null
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = v:null
let g:ale_sign_error = v:null
let g:ale_sign_info = v:null
let g:ale_sign_offset = v:null
let g:ale_sign_style_error = v:null
let g:ale_sign_style_warning = v:null
let g:ale_sign_warning = v:null
let g:ale_sign_highlight_linenrs = v:null
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:
```

**Comment by w0rp** [comment]
Can you create a GitHub repository with an example project to demonstrate the issue? I don't think I'll be able to figure this out otherwise.

**Comment by ukch** [comment]
Apologies, it seems I can no longer reproduce this issue!
